diff --git a/BUILD.bazel b/BUILD.bazel
new file mode 100644
index 0000000..801c5d1
--- /dev/null
+++ b/BUILD.bazel
@@ -0,0 +1,61 @@
+cc_library(
+    name = "libpg_query",
+    hdrs = ["pg_query.h"],
+    visibility = ["//visibility:public"],
+    deps = [":lib"],
+)
+
+proto_library(
+    name = "proto",
+    srcs = ["protobuf/pg_query.proto"],
+    visibility = ["//visibility:public"],
+)
+
+cc_library(
+    name = "lib",
+    srcs = glob([
+        "src/*.c",
+        "src/postgres/*.c",
+    ]),
+    hdrs = ["pg_query.h"] + glob([
+        "src/**/*.h",
+    ]),
+    includes = [
+        "src/include/",
+        "src/postgres/include/",
+    ],
+    textual_hdrs = glob([
+        "src/include/*.c",
+        "src/postgres/include/*.c",
+    ]),
+    deps = [
+        ":c_proto",
+        ":xxhash",
+    ],
+)
+
+cc_library(
+    name = "c_proto",
+    srcs = [
+        "protobuf/pg_query.pb-c.c",
+    ],
+    hdrs = [
+        "protobuf/pg_query.pb-c.h",
+    ],
+    includes = ["protobuf/"],
+    deps = [":protobuf-c"],
+)
+
+cc_library(
+    name = "xxhash",
+    srcs = ["vendor/xxhash/xxhash.c"],
+    hdrs = ["vendor/xxhash/xxhash.h"],
+    includes = ["vendor/"],
+)
+
+cc_library(
+    name = "protobuf-c",
+    srcs = ["vendor/protobuf-c/protobuf-c.c"],
+    hdrs = ["vendor/protobuf-c/protobuf-c.h"],
+    includes = ["vendor/"],
+)
diff --git a/MODULE.bazel b/MODULE.bazel
new file mode 100644
index 0000000..f33a236
--- /dev/null
+++ b/MODULE.bazel
@@ -0,0 +1 @@
+module(name = "libpg_query", version = "0.0.0")
